core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java

```diff
@@ -0,0 +1,198 @@
+
+    @Nonnull
+    @Override
+    public LocalDate getEnd() {
+        return END;
+    }
+
+    /**
+     * To allow asserting this info in tests.
+     * @return the events gathered.
+     */
+    @VisibleForTesting
+    /* package */ static MissingClassEvents getEvents() {
+        return events;
+    }
+
+    @CheckForNull
+    @Override
+    public JSONObject createContent() {
+        JSONObject info = new JSONObject();
+        info.put("core", Jenkins.getVersion() != null ? Jenkins.getVersion().toString() : "UNKNOWN");
+        info.put("clientDate", clientDateString());
+        info.put("classMissingEvents", formatEventsAndInitialize());
+
+        return JSONObject.fromObject(info);
+    }
+
+    /**
+     * Returns the events gathered as a Map ready to use in a Json object to send via telemetry and clean the map to
+     * gather another window of events.
+     * @return the map of missed classes events gathered along this window of java 11 telemetry
+     */
+    @Nonnull
+    private JSONArray formatEventsAndInitialize() {
+        // Save the current events and clean for next (not this one) telemetry send
+        ConcurrentHashMap<List<StackTraceElement>, MissingClassEvent> toReport = MissingClassTelemetry.events.getEventsAndClean();
+        if (LOGGER.isLoggable(Level.FINE)) LOGGER.fine("Cleaned events for missed classes on telemetry for Java 11");
+
+        return formatEvents(toReport);
+    }
+
+    /**
+     * Format the events gathered in a map used to create the json object to send via telemetry. The events are named by
+     * the class not found. But a class could be repeated if it was thrown from several places. The interesting
+     * pieces of information we want to gather are the places where the {@link ClassNotFoundException} or the
+     * {@link NoClassDefFoundError} errors happens, rather than the class itself.
+     * @param events events collected in this telemetry window.
+     * @return the events formatted in a map.
+     */
+    @Nonnull
+    private JSONArray formatEvents(@Nonnull ConcurrentHashMap<List<StackTraceElement>, MissingClassEvent> events) {
+        JSONArray jsonEvents = new JSONArray();
+
+        events.forEach((stackTrace, event) -> {
+            JSONObject eventObject = new JSONObject();
+            eventObject.put("className", event.getClassName());
+            eventObject.put("class", event.getClassName());
+            eventObject.put("time", event.getTime());
+            eventObject.put("occurrences", Long.toString(event.getOccurrences()));
+            eventObject.put("stacktrace", event.getStackTrace());
+
+            jsonEvents.add(eventObject);
+        });
+
+        return jsonEvents;
+    }
+
+    /**
+     * The current time in the same way as other telemetry implementations.
+     * @return the UTC time formatted with the pattern yyyy-MM-dd'T'HH:mm'Z'
+     */
+    @Nonnull
+    static String clientDateString() {
+        TimeZone tz = TimeZone.getTimeZone("UTC");
```

Comment:
```
return OffsetDateTime.now(ZoneOffset.UTC).toString();
```

Already ISO-8601 formatted.

Reply:
C&P I prefer to keep it like other telemetry implementations. Maybe we can make this method global in the Telemetry class or make the TelemetryReporter add the date with your proposal. WDYT @daniel-beck as the author of Telemetry.